kong/clustering/compat/checkers.lua

```diff
@@ -0,0 +1,179 @@
+local pairs  = pairs
+local ipairs = ipairs
+
+
+local log_warn_message
+do
+  local ngx_log = ngx.log
+  local ngx_WARN = ngx.WARN
+  local fmt = string.format
+
+  local KONG_VERSION = require("kong.meta").version
+
+  local _log_prefix = "[clustering] "
+
+  log_warn_message = function(hint, action, dp_version, log_suffix)
+    local msg = fmt("Kong Gateway v%s %s " ..
+                    "which is incompatible with dataplane version %s " ..
+                    "and will %s.",
+                    KONG_VERSION, hint, dp_version, action)
+    ngx_log(ngx_WARN, _log_prefix, msg, log_suffix)
+  end
+end
+
+
+local compatible_checkers = {
+  { 3003000000, --[[ 3.3.0.0 ]]
+    function(config_table, dp_version, log_suffix)
+      -- remove updated_at field for core entities ca_certificates, certificates, consumers,
+      -- targets, upstreams, plugins, workspaces, clustering_data_planes and snis
+      local entity_names = {
+        "ca_certificates", "certificates", "consumers", "targets", "upstreams",
+        "plugins", "workspaces", "clustering_data_planes", "snis",
+        }
+
+      local has_update
+      local updated_entities = {}
+
+      for _, name in ipairs(entity_names) do
+        for _, config_entity in ipairs(config_table[name] or {}) do
+          if config_entity["updated_at"] then
+
+            config_entity["updated_at"] = nil
+
+            updated_entities[name] = true
+            has_update = true
+          end
+        end
+      end
+
+      if has_update then
+        for name, _ in pairs(updated_entities) do
```

Comment:
Can we make this an array too, so that order is always the same?

Reply:
I think the order of these entities is meanless, we should check them all.